### PR TITLE
Fixes #693 Add query log on exceptions

### LINKED ISSUE #693: FromCataloguesExtractionRequestFulfiller.GetAllMatchingFiles should indicate what table was being queried on Exception
When CohortExtractor is unable to service a query it should indicate what query was running when it blew up.

### PR DESCRIPTION
**Pull request by rkm**
## Proposed Changes

Add logging of query statements to QueryToExecute on exceptions.

## Types of changes

What types of changes does your code introduce? Tick all that apply.

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation-Only Update (if none of the other choices apply)
  - In this case, ensure that the message of the head commit from the source branch is prefixed with `[skip ci]`

## Checklist

By opening this PR, I confirm that I have:

- [x] Reviewed the [contributing](https://github.com/SMI/SmiServices/blob/master/CONTRIBUTING.md) guidelines for this repository
- [x] Ensured that the PR branch is in sync with the target branch (i.e. it is automatically merge-able)
- [ ] Updated any relevant API documentation
- [ ] Created or updated any tests if relevant
- [x] Created a [news](https://github.com/SMI/SmiServices/blob/master/news/README.md) file
    -   NOTE: This ***must*** include any changes to any of the following files: default.yaml, any of the RabbitMQ server configurations, GlobalOptions.cs
- [x] Listed myself in the [CONTRIBUTORS](https://github.com/SMI/SmiServices/blob/master/CONTRIBUTORS.md) file 🚀
- [x] Requested a review by one of the repository maintainers

## Issues

- Fixes #693

### CODE REVIEW

File: news/698-bugfix.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Add logging of query statements to QueryToExecute for exceptions
```
Comment (tznind) [inline news/698-bugfix.md:1]: Maybe: 

CohortExtractor database queries that crash during execution are now logged

File: src/microservices/Microservices.CohortExtractor/Execution/RequestFulfillers/QueryToExecute.cs [hunk 2]
```diff
@@ -125,33 +126,45 @@ public IEnumerable<QueryToExecuteResult> Execute(string valueToLookup, List<IRej
             using (DbConnection con = Server.GetConnection())
             {
                 con.Open();
-                DbDataReader r = Server.GetCommand(GetSqlForKeyValue(valueToLookup), con).ExecuteReader();
```
Comment (tznind) [inline src/microservices/Microservices.CohortExtractor/Execution/RequestFulfillers/QueryToExecute.cs:128]: Its fine this time but I think we should avoid renaming locals and making whitespace changes in commits if possible as it makes it harder to review exactly what is being done in the PR.

**Review by tznind — DISMISSED** [review]

File: news/698-bugfix.md (hunk 1 above)
Comment (rkm) [reply news/698-bugfix.md]: Sure - updated.

File: src/microservices/Microservices.CohortExtractor/Execution/RequestFulfillers/QueryToExecute.cs (hunk 2 above)
Comment (rkm) [reply src/microservices/Microservices.CohortExtractor/Execution/RequestFulfillers/QueryToExecute.cs:128]: Yeah I agree - PR changes should be minimal.

Here I wanted to clarify the `r` variable. In general, I think we would benefit from applying a linter like dotnet-format automatically to eliminate these sorts of changes (whitespace, code style etc.) needing to be discussed in PRs.

**Review by tznind — APPROVED** [review]

**Pull request merged**